strapi/strapi
# Relations: error handling for non existent relations

### What does it do?

Detects non existent relations when updating an entity through `entity-service.update`

### Why is it needed?

When trying to save an entity with a Relation that doesn’t exist, the backend throw a 500 error without a descriptive error message

### How to test it?

- Create a content type (e.g. `testContent`) with a one to many relation to `addresses`
- Create some addresses
- Start to create a `testContent` entry and select some addresses in the relation field
- Before clicking save, delete the address entries in another browser tab
- Click save on the new `testContent` entry
- The API should throw an error with a useful error message

---

- Fixes https://github.com/strapi/strapi/issues/10871
